types/react-relay/test/relay-experimental-tests.tsx

```diff
@@ -957,27 +974,241 @@ function LoadQuery() {
         }
     `;
 
-    const variables: AppQueryVariables  = {
-        id: "1"
+    const variables: AppQueryVariables = {
+        id: '1',
     };
 
     const preloadedQuery = loadQuery<AppQuery>(environment, query, variables, {
         fetchPolicy: 'store-or-network',
         networkCacheConfig: {
-            force: true
-        }
+            force: true,
+        },
     });
 
-    function ShouldPassIntoUsePreloadQuery({preloadedQuery}: {preloadedQuery: PreloadedQuery<AppQuery>}) {
+    function ShouldPassIntoUsePreloadQuery({ preloadedQuery }: { preloadedQuery: PreloadedQuery<AppQuery> }) {
         const data = usePreloadedQuery(query, preloadedQuery); // $ExpectType AppQueryResponse
 
         return <>{data.user.name}</>;
     }
 
     // To depict some sort of running app
     function App() {
+        return <ShouldPassIntoUsePreloadQuery preloadedQuery={preloadedQuery} />;
+    }
+}
+
+/*
+EntryPoint tests
+ */
+
+function EntryPointTests() {
+    interface SomeQueryResponse {
+        readonly post: {
+            readonly name: string;
+        };
+    }
+
+    interface SomeQueryVariables {
+        slug: string;
+    }
+
+    interface SomeQuery {
+        readonly response: SomeQueryResponse;
+        readonly variables: SomeQueryVariables;
+    }
+
+    const query = graphql`
+        query SomeQuery($slug: String!) {
+            post {
+                name
+            }
+        }
+    `;
+
+    // RootEntrypoint.entrypoint
+    const RootEntrypointComponent: EntryPointComponent<{ someQueryRef: SomeQuery }, {}, { onClick: () => void }> = ({
+        queries,
+        props,
+    }) => {
+        const data = usePreloadedQuery(query, queries.someQueryRef);
+
+        return <button onClick={props.onClick}>{data.post.name}</button>;
+    };
+
+    interface Params {
+        slug: string;
+    }
+
+    const entrypoint: EntryPoint<typeof RootEntrypointComponent, Params> = {
+        root: JSResource<typeof RootEntrypointComponent>(),
+        getPreloadProps(
+            // $ExpectType Params
+            params,
+        ) {
+            return {
+                queries: {
+                    someQueryRef: {
+                        parameters: query as any,
```

Comment:
Why is this `as any` cast necessary?

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48638#discussion_r510080943